modules/telemetry/telemetry/include/hephaestus/telemetry/log_sink.h

```diff
@@ -0,0 +1,145 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
+
+#include <functional>
+#include <iomanip>
+#include <source_location>
+#include <sstream>
+#include <string>
+#include <string_view>
+#include <thread>
+#include <utility>
+#include <vector>
+
+namespace heph {
+enum class LogLevel : std::uint8_t { TRACE, DEBUG, INFO, WARN, ERROR, FATAL };
```

Comment:
Am I crazy if I suggest:
```suggestion
enum LogLevel : std::uint8_t { TRACE, DEBUG, INFO, WARN, ERROR, FATAL };
```
so that we can do: `heph::log(heph::INFO, ...)` 

This symbols are very log specific, I doubt we will cause any clashes

Reply:
I actually also considered it, however I am realy not a fan of implicit cast of enums to uchar...
we could consider strong typing, just defining them as individual structs and enforcing them with a concept like
https://godbolt.org/z/nG1G4PrGb
might be overkill though

Reply:
This is a cool concept, but I think we can live with having a non class enum. It is for a very specific case, so it should be ok.